packages/mui-material/src/Tooltip/Tooltip.js

```diff
@@ -669,72 +681,67 @@ const Tooltip = React.forwardRef(function Tooltip(inProps, ref) {
       tooltipModifiers = tooltipModifiers.concat(PopperProps.popperOptions.modifiers);
     }
 
+    if (resolvedPopperProps?.popperOptions?.modifiers) {
+      tooltipModifiers = tooltipModifiers.concat(resolvedPopperProps.popperOptions.modifiers);
+    }
+
     return {
       ...PopperProps.popperOptions,
+      ...resolvedPopperProps?.popperOptions,
       modifiers: tooltipModifiers,
     };
-  }, [arrowRef, PopperProps]);
-
-  const ownerState = {
-    ...props,
-    isRtl,
-    arrow,
-    disableInteractive,
-    placement,
-    PopperComponentProp,
-    touch: ignoreNonTouchEvents.current,
-  };
+  }, [arrowRef, PopperProps.popperOptions, resolvedPopperProps?.popperOptions]);
 
   const classes = useUtilityClasses(ownerState);
 
-  const PopperComponent = slots.popper ?? components.Popper ?? TooltipPopper;
-  const TransitionComponent =
-    slots.transition ?? components.Transition ?? TransitionComponentProp ?? Grow;
-  const TooltipComponent = slots.tooltip ?? components.Tooltip ?? TooltipTooltip;
-  const ArrowComponent = slots.arrow ?? components.Arrow ?? TooltipArrow;
-
-  const popperProps = appendOwnerState(
-    PopperComponent,
-    {
-      ...PopperProps,
-      ...(slotProps.popper ?? componentsProps.popper),
-      className: clsx(
-        classes.popper,
-        PopperProps?.className,
-        (slotProps.popper ?? componentsProps.popper)?.className,
-      ),
+  const externalForwardedProps = {
+    slots: {
+      popper: components.Popper,
+      transition: components.Transition ?? TransitionComponentProp,
+      tooltip: components.Tooltip,
+      arrow: components.Arrow,
+      ...slots,
+    },
+    slotProps: {
+      arrow: slotProps.arrow ?? componentsProps.arrow,
+      popper: { ...PopperProps, ...(resolvedPopperProps ?? componentsProps.popper) }, // resolvedPopperProps can be spread because it's already an object
+      tooltip: slotProps.tooltip ?? componentsProps.tooltip,
+      transition: slotProps.transition ?? { ...TransitionProps, ...componentsProps.transition },
```

Comment:
Let's keep the original merging order (as with Popper)

```suggestion
      transition: { ...TransitionProps, ...(slotProps.transition ?? componentsProps.transition) },
```

This avoids breaking changes at this stage.

Reply:
Okay, but I have to do the resolved props similar to the tooltipPopper props because `slotProps.transition` could be a function, spreading like this would throw an error.